packages/cli/scripts/pre-designed-nft.ts

```diff
@@ -0,0 +1,231 @@
+  const response = await openai.createImage({
+    prompt: prompt,
+    n: numberOfImages,
+    size: imageSize
+  })
+
+  for (let i = 0; i < numberOfImages; i++) {
+    const imageUrl = response.data.data[i].url
+    const imageResponse = await fetch(imageUrl)
+    const fileStream = fs.createWriteStream(`${storedDir}/${i}`, { flags: 'wx' })
+    await imageResponse.body.pipe(fileStream)
+  }
+}
+
+export async function uploadImagesAndMetadataToIPFS(
+  localDir: string,
+  ipfsDir: string,
+  metadataConfigFile: string,
+  infuraProjectId: string,
+  infuraProjectSecret: string
+): Promise<string> {
+  const metadataConfig = parseMetadataConfig(metadataConfigFile, localDir)
+  console.log(metadataConfig)
+
+  const files = Object.keys(metadataConfig)
+
+  const toBeUploaded = []
+  files.forEach((file) => {
+    const localFilePath = path.join(localDir, file)
+    const ipfsFilePath = path.join(ipfsDir, file)
+    const fileStream = fs.createReadStream(localFilePath)
+    toBeUploaded.push({ path: ipfsFilePath, content: fileStream })
+    return file
+  })
+
+  if (toBeUploaded.length > 0) {
+    const ipfsClient = createIPFSClient(infuraProjectId, infuraProjectSecret)
+
+    let remoteDirURL: string | undefined = undefined
+    const metadata: NFTMetadata[] = []
+    for await (const result of ipfsClient.addAll(toBeUploaded)) {
+      if (result.path === ipfsDir) {
+        remoteDirURL = `https://ipfs.io/ipfs/${result.cid.toString()}`
+      }
+    }
+
+    if (remoteDirURL) {
+      files.forEach((file, index) => {
+        metadata.push({
+          name: metadataConfig[file].name ?? `#${index}}`,
+          description: metadataConfig[file].description,
+          image: `${remoteDirURL} / ${file}`,
+          attributes: convertAttributes(metadataConfig[file].attributes)
+        })
+      })
+    }
+
+    return await uploadImageMetadataToIPFS(ipfsDir, metadata, infuraProjectId, infuraProjectSecret)
+  } else {
+    throw new Error(`Directory ${localDir} is empty`)
+  }
+}
+
+export function validateMetadataConfig(config: object, localDir: string): NFTMetadataConfig {
+  const images = Object.keys(config)
+
+  images.forEach((image) => {
+    const fileExists = fs.existsSync(path.join(localDir, image))
+    if (!fileExists) {
+      throw new Error(`File ${image} does not exist in ${localDir}`)
+    }
+
+    const metadata = config[image] as NFTMetadataConfig
+
+    Object.keys(metadata).forEach((key) => {
+      if (!validMetadataFields.includes(key)) {
+        throw new Error(`Invalid field ${key} in ${metadata}, only ${validMetadataFields} are allowed`)
+      }
+    })
+
+    validateStringTypeIfExists(image, metadata, 'name')
+    validateStringTypeIfExists(image, metadata, 'description')
+    validateAttributesTypeIfExists(image, metadata['attributes'])
+  })
+
+  return config as NFTMetadataConfig
+}
+
+export async function validateMetadataFile(tokenBaseUri: string, maxSupply: number): Promise<NFTMetadata[]> {
```

Comment:
This function is the same as `validateTokenBaseUri`.

Reply:
This is not needed any more, removed :+1: